app/presenters/outcome_presenter.rb

```diff
@@ -1,7 +1,25 @@
 require 'erubis'
 
 class OutcomePresenter < NodePresenter
-  class OutcomeTemplateMissing < StandardError; end
+  class ViewContext
+    def initialize(state)
+      @state = state
+      state_methods = state.methods - SmartAnswer::State.new(nil).methods
+      @state_reader_methods = state_methods.reject { |m| m =~ /=$/ }
```

Comment:
Hmm, I've actually got an array of symbols so I can't use `#end_with?` without first turning them to strings. Either of these work but I wonder whether they're actually any clearer now. What do you think?

    @state_reader_methods = state_methods.reject { |m| m.to_s.end_with?('=') }

    or

    @state_reader_methods = state_methods.map(&:to_s).reject { |m| m.end_with?('=') }.map(&:to_sym)